src/Hosting/DnsRecordInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * This file is part of the Acquia Platform: Cloud Data Model.
+ *
+ * (c) Acquia, Inc.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Acquia\Platform\Cloud\Hosting;
+
+/**
+ * Interface DnsRecordInterface.
+ */
+interface DnsRecordInterface
+{
+    /**
+     * Return the type of DNS record (A, AAAA, CNAME).
+     *
+     * @return string
+     *   The type of the DNS record.
+     */
+    public function getType(): string;
+
+    /**
+     * Return the value of the DNS record (IPv4 address, IPv6 address, CNAME).
+     *
+     * @return string
+     *   The value of the DNS record.
+     */
+    public function getValue(): string;
```

Comment:
I suggest changing `getValue()` to `getRData()` to match the DNS specification.

Reply:
@webbj74 none of this maps to actual DNS records or spec, this is based on the response from hosting's getDnsGuidance endpoints.

Reply:
(Which is admittedly a little confusing, but this maps to what hosting-api and Cloud API v2 are doing.)

Reply:
Changing all instances of `DnsRecord*` to `DnsConfig*` to avoid confusion.